### PR TITLE
Event page

### PR DESCRIPTION
**Pull request by CristobalMe**

### CODE REVIEW

**Review by ygongdev — COMMENTED** [review]
Remove .DS_Store and check if it is added in your .gitignore

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Ignore artifacts:
```
Comment (ygongdev) [inline .prettierignore:1]: This should be left out of this PR

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (ygongdev) [inline .prettierrc:1]: This should be left out as well.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,3 @@
 # Event_Finder
+
```
Comment (ygongdev) [inline README.md:2]: Doesn't seem to be intended change

File: backend/server.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+const SequelizeStore = SequelizeStoreInit(session.Store);
+const sessionStore = new SequelizeStore({
+  db: newSequelize,
+});
+
+// Session middleware
+app.use(
+  session({
+    secret: "your-secret-key",
```
Comment (ygongdev) [inline backend/server.js:31]: should we be providing an actual secret key?

File: backend/server.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      secure: false,
+      expires: new Date(Date.now() + 365 * 24 * 60 * 60 * 1000), // 1 year in milliseconds
+    },
+  }),
+);
+sessionStore.sync();
+
+const eventRoutes = require("./routes/eventRoutes");
+const userRoutes = require("./routes/userRoutes");
+const commentRoutes = require("./routes/commentRoutes");
```
Comment (ygongdev) [inline backend/server.js:46]: `require` imports should be moved to the top of the file to be consistent with the other imports unless this is intended and necessary

File: frontend/.eslintrc.cjs [hunk 6]
```diff
@@ -2,20 +2,20 @@ module.exports = {
...
+  settings: { react: { version: "18.2" } },
+  plugins: ["react-refresh"],
   rules: {
-    'react/jsx-no-target-blank': 'off',
-    'react-refresh/only-export-components': [
-      'warn',
+    "react/jsx-no-target-blank": "off",
+    "react-refresh/only-export-components": [
+      "warn",
       { allowConstantExport: true },
```
Comment (ygongdev) [inline frontend/.eslintrc.cjs:18]: Leave frontend changes in a separate PR. This also seems like a prettier change?

File: frontend/src/components/eventPage/EventPage.jsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  useEffect(() => {
+    if (changeInEvent) {
+      fetchEvent();
+      fetchComments();
+    }
+    setchangeInEvent(false);
+  }, [event]);
+
+  const fetchEvent = () => {
+    fetch(`http://localhost:3000${current_link}`)
```
Comment (toriqyanM) [inline frontend/src/components/eventPage/EventPage.jsx:22]: Consider storing it as an env variable so that once you deploy your server side, you don't have the risk of missing a url to change

File: frontend/src/components/eventPage/EventPage.jsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+      .then((response) => response.json())
+      .then((data) => setEvent(data))
+      .catch((error) => console.error("Error fetching:", error));
+  };
+
+  const fetchComments = () => {
+    fetch(`http://localhost:3000/comments/${current_eventId}`)
```
Comment (toriqyanM) [inline frontend/src/components/eventPage/EventPage.jsx:29]: Same as above

File: .prettierignore (hunk 1 above)
Comment (CristobalMe) [reply .prettierignore:1]: Done

File: .prettierrc (hunk 2 above)
Comment (CristobalMe) [reply .prettierrc:1]: Done

File: README.md (hunk 3 above)
Comment (CristobalMe) [reply README.md:2]: Done

File: backend/server.js (hunk 4 above)
Comment (CristobalMe) [reply backend/server.js:31]: To Do:
Delete sequelize db & pass content to prisma

File: backend/server.js (hunk 5 above)
Comment (CristobalMe) [reply backend/server.js:46]: This is intended (Based on the CodePath code & classes)

File: frontend/.eslintrc.cjs (hunk 6 above)
Comment (CristobalMe) [reply frontend/.eslintrc.cjs:18]: Yes, sorry

Comment (CristobalMe) [reply frontend/.eslintrc.cjs:18]: Done

File: frontend/src/components/eventPage/DisplayEventComments.jsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+  const toggleForm = () => {
+    setForm(!form);
+  };
+
+  // const handleNewComment = () => {
+  //     console.log("a")
+  //     toggleForm();
+  // }
+
```
[the remark this answers is not in the document]
Comment (CristobalMe) [reply frontend/src/components/eventPage/DisplayEventComments.jsx:27]: Done

File: backend/routes/commentRoutes.js
[the remark this answers is not in the document]
Comment (CristobalMe) [reply backend/routes/commentRoutes.js:1]: To Do:
Error handling in prisma db calls

File: frontend/src/App.jsx [hunk 10]
```diff
@@ -1,31 +1,63 @@
...
+// Change parameters of EventsCarousel (Needs to complete data models and user auth first)
+// Put urls in and .env file
+// Some kind of email format validation, so that entries like foo@bar can't be used as email
+// Enforce pattern matching for password, e.g must have a symbol
+// GET LOCATION OF EVENT (LAT & LONG)
+// Update event Model (Ask for date of the event)
+
+// DONE
+// Login/signup routing logic
+// Redundant code with the creation of InitialHeader
```
[the remark this answers is not in the document]
Comment (CristobalMe) [reply frontend/src/App.jsx:24]: Sorry, this wasn't a good PR

File: frontend/src/App.jsx [hunk 11]
```diff
@@ -1,31 +1,63 @@
...
+  const updateUser = (newUser) => {
+    setUser(newUser);
+  };
+
+  useEffect(() => {
+    // Save the user data to storage whenever the user state changes
+    localStorage.setItem("user", JSON.stringify(user));
+    console.log(localStorage.getItem("user"));
+    // console.log(typeof localStorage.getItem('user'));
+    // console.log(String === typeof localStorage.getItem('user') instanceof String);
```
[the remark this answers is not in the document]
Comment (CristobalMe) [reply frontend/src/App.jsx:40]: Done

File: frontend/src/components/carousel/EventsCarousel.jsx [hunk 12]
```diff
@@ -34,17 +34,22 @@ const Carousel = () => {
     ],
   };
 
+  let data = [];
+  if (events.events) {
+    data = events.events;
+  }
+
```
[the remark this answers is not in the document]
Comment (CristobalMe) [reply frontend/src/components/carousel/EventsCarousel.jsx:41]: Done

File: frontend/src/components/eventPage/EventPage.jsx [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+      .then((response) => response.json())
+      .then((data) => setComments(data))
+      .catch((error) => console.error("Error fetching:", error));
+  };
+
+  //console.log(event)
+  //console.log(comments)
```
[the remark this answers is not in the document]
Comment (CristobalMe) [reply frontend/src/components/eventPage/EventPage.jsx:36]: Done

File: frontend/src/components/eventPage/EventPage.jsx [hunk 14]
```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect } from "react";
+import Header from "../header/Header";
+import DisplayEventInfo from "./DisplayEventInfo";
+import DisplayEventComments from "./DisplayEventComments";
+
+const EventPage = () => {
+  const current_link = window.location.pathname;
+  const current_eventId = current_link.split("/")[2];
```
[the remark this answers is not in the document]
Comment (CristobalMe) [reply frontend/src/components/eventPage/EventPage.jsx:8]: To Do:
get id through useParams

File: frontend/src/components/eventPage/EventPage.jsx (hunk 7 above)
Comment (CristobalMe) [reply frontend/src/components/eventPage/EventPage.jsx:22]: To Do:
Store urls in a .env

File: frontend/src/components/eventPage/EventPage.jsx (hunk 8 above)
Comment (CristobalMe) [reply frontend/src/components/eventPage/EventPage.jsx:29]: To Do:
Store urls in a .env
